### ISSUE TITLE
Support for non-toolbox projects that use toolboxes

### ISSUE THREAD

**Issue by benjamin-heasly** [ToolboxHub/ToolboxToolbox#35]
Currently, ToolboxToolbox is aimed at deploying shared toolboxes.  These are usually:
 - fairly stable
 - expected to be used by anyone
 - deployed into a root "toolboxes" folder
 - known to the ToolboxRegistry
 - deployed as one-liners with `tbUse()` or `tbDeployToolboxes()`

It would also be useful to support "projects", which may depend on shared toolboxes, but are developed and used a bit differently.  In this context, a project might be:
 - frequently revised, refactored, work in progress
 - expected to be used by one person or small groups who are "in the know"
 - deployed to folders outside the root "toolboxes" folder
 - not known to the ToolboxRegistry
 - deployed using means other than or in addition to `tbUse()` and `tbDeployToolboxes()`

We should add a `tbUseProject()` utility to support workflows that include a mix of shared toolboxes and projects, as described above.

Details in comments to follow.

**Comment by benjamin-heasly** [comment]
Here is a proposed workflow for a user with a "project" and `tbUseProject()`:

Configure a root "projects" folder for ToolboxToolbox.  This would be a new line in `startup.m`.

Create a new project "Foo" or obtain an existing project repository for "Foo".  Place it in a folder named Foo, anywhere inside the configured projects folder.

The project should contain a configuration file named `Foo.json` which contains the toolbox configuration for the Foo project.  This would be similar to the configuration files in the ToolboxRegistry.  It should declare:
 - the project Foo itself, so that Foo can be configured and/or updated
 - toolbox dependencies for Foo, so that these can be deployed along with Foo

To start using the project in Matlab, the user should type `tbUseProject('Foo')'.

In summary, the workflow for projects would be similar to the workflow for shared toolboxes.  Except:
 - initially, the use must manually create or obtain the project
 - the project toolbox configuration would be located with the project, not in the ToolboxRegistry
 - the use would type 'tbUseProject('Foo')` instead of 'tbUse('Foo')`

**Comment by benjamin-heasly** [comment]
Here are some implementation notes.

`tbUseProject('Foo')` will require a recursive search of the configured projects folder.  The criteria for a match on "Foo" would be:
 - found a "Foo.json"
 - "Foo.json" is contained directly or indirectly in a folder named "Foo".

We could add additional match constraints if necessary.  For example, we might want to require a specific subfolder location for "Foo.json", like `Foo/configuration/Foo.json`.  I am not sure how strict we need to be.

We should make the recursive search useful to users.  For example, local hook functions might need to detect where a project is located.  So we should break out a utility `tbLocateProject()`.  This would parallel other "locate" functions proposed in #29 .

I think we can implement `tbUseProject()` as `tbLocateProject()` plus `tbDeployToolboxes()`:
 - First, locate the project folder Foo and config file Foo.json
 - Then deploy, specifying the explicit path to the config file and overriding the `toolboxRoot` field of the Foo record, to point to the parent folder of Foo

**Comment by benjamin-heasly** [comment]
Closed by #51
